redbot/core/utils/views.py

```diff
@@ -0,0 +1,146 @@
+        """Format the keys to be used on a long discord.TextInput format"""
+        if keys is not None:
+            ret = ""
+            for k, v in keys.items():
+                if v:
+                    ret += f"{k} {v}\n"
+                else:
+                    ret += f"{k} YOUR_{k.upper()}\n"
+            return ret
+        else:
+            return None
+
+    async def on_submit(self, interaction: discord.Interaction):
+        if (
+            await self.bot.is_owner(interaction.user) is False
+        ):  # Prevent non-bot owners from somehow aquiring and saving the modal.
+            return await interaction.response.send_message(
+                _("This modal is for bot owners only. Whoops!"), ephemeral=True
+            )
+
+        if self.default_keys is not None:
+            converter = get_dict_converter(*self.default_keys, delims=[";", ",", " "])
+        else:
+            converter = get_dict_converter(delims=[";", ",", " "])
+        tokens = " ".join(self.token_input.value.split("\n")).rstrip()
+
+        try:
+            tokens = await converter().convert(None, tokens)
+        except BadArgument as exc:
+            return await interaction.response.send_message(
+                _("{error_message}\nPlease try again.").format(error_message=str(exc)),
+                ephemeral=True,
+            )
+
+        if self.default_service is not None:  # Check is there is a service set.
+            await self.bot.set_shared_api_tokens(self.default_service, **tokens)
+            return await interaction.response.send_message(
+                _("`{service}` API tokens have been set.").format(service=self.default_service),
+                ephemeral=True,
+            )
+        else:
+            service = self.service_input.value.lower()
+            await self.bot.set_shared_api_tokens(service, **tokens)
+            return await interaction.response.send_message(
+                _("`{service}` API tokens have been set.").format(service=service),
+                ephemeral=True,
+            )
+
+
+class SetApiView(discord.ui.View):
+    def __init__(
+        self,
+        bot: Red,
+        default_service: Optional[str] = None,
+        default_keys: Optional[Union[str, List[str]]] = None,
+    ):
+        self.bot: Red = bot
+        self.default_service = default_service
+        self.default_keys = default_keys
+        super().__init__()
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        if await self.bot.is_owner(interaction.user) is False:
+            await interaction.response.send_message(
+                _("This button is for bot owners only. Oh well"), ephemeral=True
+            )
+            return False
+        return True
+
+    @discord.ui.button(
+        label=_("Set API token"),
+        style=discord.ButtonStyle.grey,
+    )
+    async def auth_button(self, button: discord.Button, interaction: discord.Interaction):
```

Comment:
Just your luck - interaction parameter order changed requiring the interaction before the button now, lol.